persist/fs/read_write_test.go

```diff
@@ -70,18 +71,44 @@ func readTestData(t *testing.T, r FileSetReader, shard uint32, timestamp time.Ti
 	require.Equal(t, len(entries), r.Entries())
 	require.Equal(t, 0, r.EntriesRead())
 
+	bloomFilter, err := r.ReadBloomFilter()
+	assert.NoError(t, err)
+	// Make sure the bloom filter doesn't always return true
+	assert.False(t, bloomFilter.Test([]byte("some_random_data")))
+	expectedM, expectedK := bloom.EstimateFalsePositiveRate(
+		uint(len(entries)), defaultIndexBloomFilterFalsePositivePercent)
+	assert.Equal(t, expectedK, bloomFilter.K())
+	// EstimateFalsePositiveRate always returns at least 1, so skip this check
+	// if len entries is 0
+	if len(entries) > 0 {
+		assert.Equal(t, expectedM, bloomFilter.M())
+	}
+
 	for i := 0; i < r.Entries(); i++ {
-		id, data, checksum, err := r.Read()
+		idFromRead, data, checksumFromRead, err := r.Read()
```

Comment:
nit: It might be best to wrap this up so that the defer's are executed inside the loop, just to avoid tests using tons of memory.  It's fine that it will be a little slower because of the fn declaration, which will hopefully be inlined.

i.e.
```go
for i := 0; i < r.Entries(); i++ {
  func() {
    x := ...
    defer x.Finalize()

    y := ...
    defer y.Finalize()
    
    ...
  }()
}
```